src/main/java/org/jenkinsci/plugins/saml/user/LoginDetailsProperty.java

```diff
@@ -138,19 +140,26 @@ protected void failedToAuthenticate(@javax.annotation.Nonnull String username) {
 
         @Override
         protected void loggedIn(@javax.annotation.Nonnull String username) {
-            SecurityRealm realm = Jenkins.getInstance().getSecurityRealm();
-            if (!(realm instanceof SamlSecurityRealm)) {
-                return;
+
+            Jenkins j = Jenkins.getInstanceOrNull();
```

Comment:
Jenkins.getInstanceOrNull() is not the replacement of Jenkins.getInstance(), it should be Jenkins.get()